cpp/validate_harvested_records.cc

```diff
@@ -281,13 +312,15 @@ bool RecordMeetsExpectations(const MARC::Record &record, const std::string &jour
             continue;   // we only care about required fields that are missing
 
         const auto equivalent_tag(EQUIVALENT_TAGS_MAP.find(field_info.name_));
-        if (seen_tags.find(field_info.name_) != seen_tags.end())
+        if (seen_tags_and_subfield_codes.find(field_info.name_ + std::string(1, field_info.subfield_code_)) != seen_tags_and_subfield_codes.end())
```

Comment:
The tests in this function should depend on whether we have a review record or a regular_article.